### PR TITLE
gh #3 Register callback Enable

### PR DESCRIPTION
**Pull request by ranjithkumarkp**
Added support for Register callback functions in L1 test cases, earlier it was disabled

### CODE REVIEW

File: src/test_l1_tvSettings.c [hunk 1]
```diff
@@ -307,13 +307,12 @@ void test_l1_tvSettings_negative_TvTerm (void)
  * | Variation / Step | Description | Test Data | Expected Result | Notes |
  * | :-------: | ------------- | --------- | --------------- | ----- |
  * | 01 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 02 | Call RegisterVideoFormatChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoFormatCallbackData& | void | tvERROR_NONE | Should Pass |
+ * | 02 | Call RegisterVideoFormatChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoFormatCallbackData * | void | tvERROR_NONE | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:310]: Remove void, already test data is added as tvVideoFormatCallbackData *

File: src/test_l1_tvSettings.c [hunk 2]
```diff
@@ -366,12 +354,13 @@ void test_l1_tvSettings_positive_RegisterVideoFormatChangeCB (void)
  * **Test Procedure:**@n
  * | Variation / Step | Description | Test Data | Expected Result | Notes |
  * | :-------: | ------------- | --------- | --------------- | ----- |
- * | 01 | call RegisterVideoFormatChangeCB() -  Register the callback function even before TvInit() | tvVideoFormatCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 01 | call RegisterVideoFormatChangeCB() -  Register the callback function even before TvInit() | tvVideoFormatCallbackData * | void | tvERROR_INVALID_STATE | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:357]: Remove void, already test data is added as tvVideoFormatCallbackData *

File: src/test_l1_tvSettings.c [hunk 3]
```diff
@@ -366,12 +354,13 @@ void test_l1_tvSettings_positive_RegisterVideoFormatChangeCB (void)
...
  * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 03 | call RegisterVideoFormatChangeCB() -  Register using a NULL data handle | tvVideoFormatCallbackData&(NULL, tvVideoFormatChangeCB) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 04 | call RegisterVideoFormatChangeCB() -  Register using a NULL callback function | tvVideoFormatCallbackData&(void*, NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 05 | call TvTerm() - Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 06 | call RegisterVideoFormatChangeCB() -  Register the callback function even after TvInit() | tvVideoFormatCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 03 | call RegisterVideoFormatChangeCB() -  Register using a NULL pointer | RegisterVideoFormatChangeCB(NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:359]: Remove the void related information, Already test data is added as NULL. 

In below, all the steps remove void which is added extra.

File: src/test_l1_tvSettings.c [hunk 4]
```diff
@@ -427,13 +421,12 @@ void test_l1_tvSettings_negative_RegisterVideoFormatChangeCB (void)
  * | Variation / Step | Description | Test Data | Expected Result | Notes |
  * | :-------: | ------------- | --------- | --------------- | ----- |
  * | 01 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 02 | Call RegisterVideoContentChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoContentCallbackData& | void | tvERROR_NONE | Should Pass |
+ * | 02 | Call RegisterVideoContentChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoContentCallbackData * | void | tvERROR_NONE | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:424]: Remove void test data, which is not required

File: src/test_l1_tvSettings.c [hunk 5]
```diff
@@ -485,12 +468,13 @@ void test_l1_tvSettings_positive_RegisterVideoContentChangeCB (void)
...
+ * | 01 | call RegisterVideoContentChangeCB() -  Register the callback function even before TvInit() | tvVideoContentCallbackData * | void | tvERROR_INVALID_STATE | | Should Pass || Should Pass |
  * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 03 | call RegisterVideoContentChangeCB() -  Register using a NULL data handle | tvVideoContentCallbackData&(NULL, tvVideoFormatChangeCB) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 04 | call RegisterVideoContentChangeCB() -  Register using a NULL callback function | tvVideoContentCallbackData&(void*, NULL) | void | tvERROR_INVALID_PARA | Should Pass |
- * | 05 | call TvTerm() - Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 06 | call RegisterVideoContentChangeCB() -  Register the callback function even after TvInit() | tvVideoContentCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 03 | call RegisterVideoContentChangeCB() -  Register using a NULL pointer | RegisterVideoContentChangeCB(NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:473]: Remove the void related information, Already test data is added as NULL.

In below, all the steps remove void which is added extra.

File: src/test_l1_tvSettings.c [hunk 6]
```diff
@@ -546,13 +535,12 @@ void test_l1_tvSettings_negative_RegisterVideoContentChangeCB (void)
  * | Variation / Step | Description | Test Data | Expected Result | Notes |
  * | :-------: | ------------- | --------- | --------------- | ----- |
  * | 01 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 02 | Call RegisterVideoResolutionChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoResolutionCallbackData& | void | tvERROR_NONE | Should Pass |
+ * | 02 | Call RegisterVideoResolutionChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoResolutionCallbackData * | void | tvERROR_NONE | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:538]: Remove void which is not required

File: src/test_l1_tvSettings.c [hunk 7]
```diff
@@ -604,12 +582,13 @@ void test_l1_tvSettings_positive_RegisterVideoResolutionChangeCB (void)
...
+ * | 01 | call RegisterVideoResolutionChangeCB() -  Register the callback function even before TvInit() | tvVideoResolutionCallbackData * | void | tvERROR_INVALID_STATE | | Should Pass |
  * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 03 | call RegisterVideoResolutionChangeCB() -  Register using a NULL data handle | tvVideoResolutionCallbackData&(NULL, tvVideoFormatChangeCB) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 04 | call RegisterVideoResolutionChangeCB() -  Register using a NULL callback function | tvVideoResolutionCallbackData&(void*, NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 05 | call TvTerm() - Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 06 | call RegisterVideoResolutionChangeCB() -  Register the callback function even after TvInit() | tvVideoResolutionCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 03 | call RegisterVideoResolutionChangeCB() -  Register using a NULL pointer | RegisterVideoResolutionChangeCB(NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:587]: Remove the void related information, Already test data is added as NULL.

In below, all the steps remove void which is added extra.

File: src/test_l1_tvSettings.c [hunk 8]
```diff
@@ -724,12 +697,13 @@ void test_l1_tvSettings_positive_RegisterVideoFrameRateChangeCB (void)
...
+ * | 01 | call RegisterVideoFrameRateChangeCB() -  Register the callback function even before TvInit() | tvVideoFrameRateCallbackData * | void | tvERROR_INVALID_STATE | Should Pass |
  * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 03 | call RegisterVideoFrameRateChangeCB() -  Register using a NULL data handle | tvVideoFrameRateCallbackData&(NULL, tvVideoFormatChangeCB) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 04 | call RegisterVideoFrameRateChangeCB() -  Register using a NULL callback function | tvVideoFrameRateCallbackData&(void*, NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 05 | call TvTerm() - Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 06 | call RegisterVideoFrameRateChangeCB() -  Register the callback function even after TvInit() | tvVideoFrameRateCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 03 | call RegisterVideoFrameRateChangeCB() -  Register using a NULL pointer | RegisterVideoFrameRateChangeCB(NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:702]: Remove the void related information, Already test data is added as NULL.

In below, all the steps remove void which is added extra.

File: src/test_l1_tvSettings.c [hunk 9]
```diff
@@ -666,13 +650,12 @@ void test_l1_tvSettings_negative_RegisterVideoResolutionChangeCB (void)
  * | Variation / Step | Description | Test Data | Expected Result | Notes |
  * | :-------: | ------------- | --------- | --------------- | ----- |
  * | 01 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 02 | Call RegisterVideoFrameRateChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoFrameRateCallbackData& | void | tvERROR_NONE | Should Pass |
+ * | 02 | Call RegisterVideoFrameRateChangeCB() - Register with the obtained data handle and a valid callback function | tvVideoFrameRateCallbackData * | void | tvERROR_NONE | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:653]: Remove void test data

**Review by Anbukannadhasan — COMMENTED** [review]
Added few review comments

File: src/test_l1_tvSettings.c (hunk 1 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 2 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 3 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 4 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 5 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 6 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 7 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 8 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

File: src/test_l1_tvSettings.c (hunk 9 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c]: Done

**Review by Anbukannadhasan — APPROVED** [review]

File: src/test_l1_tvSettings.c [hunk 10]
```diff
@@ -366,12 +354,13 @@ void test_l1_tvSettings_positive_RegisterVideoFormatChangeCB (void)
...
  * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 03 | call RegisterVideoFormatChangeCB() -  Register using a NULL data handle | tvVideoFormatCallbackData&(NULL, tvVideoFormatChangeCB) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 04 | call RegisterVideoFormatChangeCB() -  Register using a NULL callback function | tvVideoFormatCallbackData&(void*, NULL) | void | tvERROR_INVALID_PARAM | Should Pass |
- * | 05 | call TvTerm() - Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 06 | call RegisterVideoFormatChangeCB() -  Register the callback function even after TvInit() | tvVideoFormatCallbackData& | void | tvERROR_INVALID_STATE | Should Pass |
+ * | 03 | call RegisterVideoFormatChangeCB() -  Register using a NULL pointer | RegisterVideoFormatChangeCB(NULL) | tvERROR_INVALID_PARAM | Should Pass |
+ * | 04 | call RegisterVideoFormatChangeCB() -  Register using a callback data with NULL callback function | tvVideoFormatCallbackData(void*, NULL) | tvERROR_INVALID_PARAM | Should Pass |
```
Comment (srinivasgtl) [inline src/test_l1_tvSettings.c:360]: You need to call the RegisterVideoFormatChangeCB with Callback as NULL and not tvVideoFormatCallbackData.  Should be in similar lines for the step 4, 5, and 7.

File: src/test_l1_tvSettings.c [hunk 11]
```diff
@@ -390,21 +379,26 @@ void test_l1_tvSettings_negative_RegisterVideoFormatChangeCB (void)
 	result = TvInit();
 	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
-	/* Step 03: Calling tvsettings RegisterVideoFormatChangeCB and expecting the API to return success*/
+	 /* Step 03: Calling tvsettings RegisterVideoFormatChangeCB and expecting the API to return success*/
```
Comment (srinivasgtl) [inline src/test_l1_tvSettings.c:382]: and expecting the API to return the appropriate error code, not success.

File: src/test_l1_tvSettings.c [hunk 12]
```diff
@@ -332,19 +331,8 @@ void test_l1_tvSettings_positive_RegisterVideoFormatChangeCB (void)
 	callbackData.cb = tvVideoFormatChangeHandler;
```
Comment (srinivasgtl) [inline src/test_l1_tvSettings.c:331]: how about the callbackData.userdata?  what if it is NULL? Should be considered all through the file in the similar scenarios.

File: src/test_l1_tvSettings.c [hunk 13]
```diff
@@ -390,21 +379,26 @@ void test_l1_tvSettings_negative_RegisterVideoFormatChangeCB (void)
...
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
-	/* Step 05: Calling tvsettings initialization and expecting the API to return success */
+	/* Step 06: Calling tvsettings initialization and expecting the API to return success */
 	result = TvTerm();
 	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
-	/* Step 06: Calling tvsettings RegisterVideoFormatChangeCB and expecting the API to return success*/
+	/* Step 07: Calling tvsettings RegisterVideoFormatChangeCB and expecting the API to return success*/
 	callbackData.cb = tvVideoFormatChangeHandler;
```
Comment (srinivasgtl) [inline src/test_l1_tvSettings.c:402]: how about userData here? It should be set to valid value.

**Review by srinivasgtl — COMMENTED** [review]
Please update all through the changes with the present comments.

**Review by ranjithkumarkp — COMMENTED** [review]
addressed review comments

File: src/test_l1_tvSettings.c (hunk 12 above)
Comment (srinivasgtl) [reply src/test_l1_tvSettings.c:331]: As discussed, please update this. As NULL is verified in negative scenario, we should make sure that it is holding a valid info in other scenarios.

File: src/test_l1_tvSettings.c (hunk 13 above)
Comment (srinivasgtl) [reply src/test_l1_tvSettings.c:402]: should be valid data in non negative cases.

File: src/test_l1_tvSettings.c (hunk 12 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c:335]: Done

File: src/test_l1_tvSettings.c (hunk 13 above)
Comment (ranjithkumarkp) [reply src/test_l1_tvSettings.c:411]: Done

**Review by Anbukannadhasan — APPROVED** [review]

**Pull request merged**